ControlRoomApplication/ControlRoomApplication/Controllers/PLCCommunication/PLCDrivers/ProductionPLCDriver.cs

```diff
@@ -578,6 +599,12 @@ public override ushort getregvalue(ushort adr)
             // check against weather station reading
             double weatherStationTemp = Parent.WeatherStation.GetOutsideTemp();
```

Comment:
Here, we're using `Parent.WeatherStation` to get the weather station stuff, which you can do in all your uses of your `weatherStation` field!